react/components/MiniCartContent.js

```diff
@@ -70,29 +81,72 @@ class MiniCartContent extends Component {
   onUpdateItems = (id, quantity) => {
     this.setState({ showSpinner: true })
     const {
-      data: { orderForm, updateAndRefetchOrderForm },
+      data: { orderForm, updateAndRefetchOrderForm, updateOrderForm },
     } = this.props
-    const itemPayload = orderForm.items.find(item => item.id === id)
-    const index = orderForm.items.indexOf(itemPayload)
-    const updatedItem = [itemPayload].map(item => {
-      return {
-        id: parseInt(item.id),
-        index: index,
-        quantity: quantity,
-        seller: 1,
-      }
-    })
 
-    updateAndRefetchOrderForm({
-      variables: {
-        orderFormId: orderForm.orderFormId,
-        items: updatedItem,
-      },
-    }).then(() => {
-      this.setState({
-        showSpinner: false,
+    const items = this.getNoRepetedItems()
+    const itemPayloadConcatenated = items.find(item => item.id === id)
+    const itemsPayload = orderForm.items.filter(item => item.id === id)
+    const index = orderForm.items.indexOf(itemsPayload[0])
```

Comment:
use `head` of ramda.

U r using `itemsPayload[0]` many times in the code, better to assign to a variable.